Source/WebKit/UIProcess/API/Cocoa/WKFrameInfo.mm

```diff
@@ -109,4 +109,16 @@ - (BOOL)_isLocalFrame
     return _frameInfo->isLocalFrame();
 }
 
+- (BOOL)_isFocused
+{
+    auto page = _frameInfo->page();
+    if (!page)
+        return NO;
+
+    auto focusedFrame = page->focusedFrame();
+    if (!focusedFrame)
+        return _frameInfo->isMainFrame();
+    return focusedFrame->frameID() == _frameInfo->handle()->frameID();
```

Comment:
Adding a bool to FrameInfoData and getting this value from the web content process instead would allow us to check state consistency between web processes's state controllers.

I'm not sure if we also want a way to check UI process state in tests.